crates/core/src/node/p2p_impl.rs

```diff
@@ -223,39 +223,31 @@ impl NodeP2P {
         // Prepare session adapter channel for actor-based client management
         let (session_tx, session_rx) = tokio::sync::mpsc::channel(1000);
 
-        // Install session adapter in contract handler if migration enabled
-        let result_router_tx = if config.config.actor_clients {
-            ch_outbound.with_session_adapter(session_tx.clone());
-
-            // Create result router channel for dual-path result delivery
-            let (result_router_tx, result_router_rx) = tokio::sync::mpsc::channel(1000);
-
-            // Spawn Session Actor
-            use crate::client_events::session_actor::SessionActor;
-            let session_actor = SessionActor::new(session_rx, cli_response_sender.clone());
-            GlobalExecutor::spawn(async move {
-                tracing::info!("Session actor starting");
-                session_actor.run().await;
-                tracing::warn!("Session actor stopped");
-            });
-
-            // Spawn ResultRouter task
-            use crate::client_events::result_router::ResultRouter;
-            let router = ResultRouter::new(result_router_rx, session_tx.clone());
-            GlobalExecutor::spawn(async move {
-                tracing::info!("Result router starting");
-                router.run().await;
-                tracing::warn!("Result router stopped");
-            });
-
-            tracing::info!(
-                "Actor-based client management infrastructure installed with result router"
-            );
-            Some(result_router_tx)
-        } else {
-            tracing::debug!("Actor-based client management disabled");
-            None
-        };
+        // Install session adapter in contract handler
+        ch_outbound.with_session_adapter(session_tx.clone());
```

Comment:
Maybe we can do this earlier so ee dont have to relay on optional fields? Likewise elsewhere.